k2/csrc/fsa_util.cc

```diff
@@ -6,9 +6,91 @@
 
 #include "k2/csrc/fsa_util.h"
 
+#include <algorithm>
+#include <cctype>
+#include <cstdlib>
 #include <utility>
 #include <vector>
 
+#include "glog/logging.h"
+
+namespace {
+
+int32_t StringToInt(const std::string &s, bool *is_ok = nullptr) {
+  CHECK_EQ(s.empty(), false);
+
+  bool ok = false;
+  char *p = nullptr;
+  // std::strtol requires a `long` type
+  long n = std::strtol(s.c_str(), &p, 10);
+  if (*p == 0) ok = true;
+
+  int32_t res = n;
+  if (n != res) ok = false;  // out of range
+  if (is_ok != nullptr) *is_ok = ok;
+
+  return res;
+}
+
+std::vector<int32_t> StringVectorToIntVector(
+    const std::vector<std::string> &in) {
+  std::vector<int32_t> res;
+  res.reserve(in.size());
+  for (const auto &s : in) {
+    bool ok = false;
+    auto n = StringToInt(s, &ok);
+    CHECK(ok);
+    res.push_back(n);
+  }
+  return res;
+}
+
+// trim leading and trailing spaces of a string
+void TrimString(std::string *s) {
+  CHECK_NOTNULL(s);
+  auto not_space = [](int c) { return std::isspace(c) == false; };
+
+  s->erase(s->begin(), std::find_if(s->begin(), s->end(), not_space));
+  s->erase(std::find_if(s->rbegin(), s->rend(), not_space).base(), s->end());
+}
+
+// split a string to a vector of string using a set of delimiters.
+//
+// Example usage:
+//
+// @code
+//  std::string in = "1 2 3";
+//  const char* delim = " \t";
+//  std::vector<std::string> out;
+//  SplitStringToVector(in, delim, &out);
+// @encode
+void SplitStringToVector(const std::string &in, const char *delim,
+                         std::vector<std::string> *out) {
+  CHECK_NOTNULL(delim);
+  CHECK_NOTNULL(out);
+  out->clear();
+  size_t start = 0;
+  size_t end = in.size();
+  while (true) {
+    auto pos = in.find_first_of(delim, start);
+    if (pos == in.npos) break;
+
+    auto sub = in.substr(start, pos - start);
+    start = pos + 1;
+
+    TrimString(&sub);
+    if (sub.empty() == false) out->emplace_back(std::move(sub));
```

Comment:
really suggest `if(!sub.empty())` It's more readable

Reply:
done.